sdk-base/src/main/java/com/mapbox/maps/util/MapboxExceptions.kt

```diff
@@ -0,0 +1,43 @@
+package com.mapbox.maps.util
+
+/**
+ * Mapbox exception class flagging map related [RuntimeException] did occur.
+ */
+class MapboxMapException(
+  exceptionText: String?
+) : RuntimeException(exceptionText)
+
+/**
+ * Mapbox exception class flagging view annotation related [RuntimeException] did occur.
+ */
+class MapboxViewAnnotationException(
+  exceptionText: String?
```

Comment:
nit: provide default value to null

Reply:
why do we need that? it will actually require marking for Java usage. We're using nullable arg here as runtime exception allows both nullable and non-nullable exception text

Reply:
The Java `RuntimeException` has an [empty constructor](https://developer.android.com/reference/java/lang/RuntimeException#RuntimeException()) as well, so I guess we can add default value and use JvmOverloads annotation.

Reply:
@pengdev it will make sense if we allow to inherit from our classes. However we do not declare those classes as `open` meaning they're final. Thus we can design those classes as we wish and I see no sense in providing several constructors here.